pylint/checkers/classes.py

```diff
@@ -960,6 +960,22 @@ def _check_unused_private_attributes(self, node: astroid.ClassDef) -> None:
         for assign_attr in node.nodes_of_class(astroid.AssignAttr):
             if not is_attr_private(assign_attr.attrname):
                 continue
+
+            # Logic for checking false positive when using __new__,
+            # Get the returned object names of the __new__ magic function
+            # Then check if the attribute was consumed in other instance methods
+            acceptable_obj_names = ["self"]
+            if (
+                isinstance(assign_attr.scope(), astroid.FunctionDef)
+                and assign_attr.scope().name == "__new__"
+            ):
+                acceptable_obj_names.extend(
+                    [
+                        n.value.name
```

Comment:
This could potentially raise an exception if `astroid.Return.value` isn't of type `astroid.Name`. Might be a good idea to add an `isinstance` check for it.

Furthermore, what do you think about renaming `n` to something like `return_node`?